File: app/src/main/java/com/example/data/Task.kt
package com.example.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "tasks")
data class Task(
    @PrimaryKey val id: String,
    val title: String,
    val description: String,
    val category: String, // "Study" | "Work" | "Personal" | "Finance" | "Health" | "Business" | "Other"
    val deadline: String, // ISO 8601 string, e.g. "2026-06-23T15:00:00"
    val estimatedMinutes: Int,
    val importance: String, // "Low" | "Medium" | "High" | "Critical"
    val status: String, // "Not Started" | "In Progress" | "Blocked" | "Completed"
    val energyRequired: String, // "Low" | "Medium" | "High"
    val notes: String? = null,
    val createdAt: String,
    val updatedAt: String
)

@Entity(tableName = "habits")
data class Habit(
    @PrimaryKey val id: String,
    val title: String,
    val isCompleted: Boolean,
    val date: String // e.g. "2026-06-23"
)

@Entity(tableName = "settings")
data class Setting(
    @PrimaryKey val key: String,
    val value: String
)

File: web/src/types/task.ts
export interface Task {
  id: string;
  title: string;
  description: string;
  category: 'Study' | 'Work' | 'Personal' | 'Finance' | 'Health' | 'Business' | 'Other';
  deadline: string; // ISO date-time string
  estimatedMinutes: number;
  importance: 'Low' | 'Medium' | 'High' | 'Critical';
  status: 'Not Started' | 'In Progress' | 'Blocked' | 'Completed';
  energyRequired: 'Low' | 'Medium' | 'High';
  notes?: string;
  
  // Risk Engine output fields
  riskScore?: number;
  urgencyLevel?: 'Safe' | 'Watch' | 'Urgent' | 'Critical' | 'Overdue';
  riskReason?: string;
  timeRemainingText?: string;
  recommendedAction?: string;
  rescueRequired?: boolean;
  microSteps?: MicroStep[];
}

export interface MicroStep {
  id: string;
  title: string;
  estimatedMinutes: number;
  priority: number;
  isCompleted: boolean;
}

export interface Habit {
  id: string;
  title: string;
  isCompleted: boolean;
  date: string;
}

export interface Message {
  id: string;
  text: string;
  sender: 'user' | 'coach';
  timestamp: string;
}
